test/e2e/state.js

```diff
@@ -0,0 +1,141 @@
+const fs = require("fs");
+
+const defaultState = {
+  // These don't work quite perfectly yet, there are still places where we are
+  // hardcoding localhost:8080, in files like `header.js` and `runTest.js`.  When
+  // we move to Vercel branch preview testing, this work will have to be completed
+  EXAMPLE_SERVER: "http://localhost:8080/test/examples",
+  REPLAY_SERVER: "http://localhost:8080",
+
+  // This API key allows access to the whole team that holds all these test
+  // recordings. If someone wanted to, they could go in and delete the workspace
+  // or recordings in it or anything. While thats not great, it's also not
+  // the end of the world. If someone does that, we can always change
+  // this code to only run in CI in the main repo and have this as a secret.
+  // It's a lot easier to hardcode it for now though.
+  replayApiKey: "rwk_7XPbO5fhz0bkhANYXtN2dkm74wNQCchXf2OxVgAerTQ",
+  DefaultDispatchServer: "wss://dispatch.replay.io",
+
+  usePlaywright: true,
+  ExampleRecordings: fs.existsSync("./test/example-recordings.json")
+    ? JSON.parse(fs.readFileSync("./test/example-recordings.json"))
+    : {},
+
+  patterns: [],
+  stripeIndex: undefined,
+  stripeCount: undefined,
+  dispatchServer: undefined,
+  shouldRecordExamples: false,
+  shouldRecordViewer: false,
+  recordExamplesSeparately: false,
+  testTimeout: 240,
+  skippedTests: process.env.SKIPPED_TESTS,
+  onlyTarget: undefined,
+
+  // Runtime state
+  count: 1,
+  startTime: Date.now(),
+  timings: {},
+  failures: [],
+
+  // CONSTANTS
+
+  // Creating an example returns this constant if the test should be skipped.
+  SkipTest: "skipped",
+
+  // Creating an example returns this constant if the test should be recorded
+  // as part of the viewer process. The test will be recorded and then stopped to
+  // switch to the viewer, as when using replay normally.
+  RecordExampleInViewer: "record-example-in-viewer",
```

Comment:
I'm not sure I understand this one. Why do we have a constant for this rather than doing something like `clickOn(viewerButton)` in the tests that need it?

Reply:
yeah - i didn't follow this either. Lets see if we still need it.